extension/src/app.rs

```diff
@@ -0,0 +1,247 @@
+use crate::{
+    components::{CreateWallet, TradeInfo, UnlockWallet, WalletDetails},
+    event_bus::EventBus,
+    wallet_updater::WalletUpdater,
+};
+use js_sys::Promise;
+use message_types::bs_ps::{self, BackgroundStatus, TransactionData, WalletStatus};
+use serde::{Deserialize, Serialize};
+use wasm_bindgen::prelude::*;
+use wasm_bindgen_extension::browser;
+use wasm_bindgen_futures::{spawn_local, JsFuture};
+use yew::prelude::*;
+
+// We do not support renaming the wallet for now
+pub const WALLET_NAME: &str = "demo-wallet";
+
+pub struct App {
+    link: ComponentLink<Self>,
+    state: State,
+    _event_bus: Box<dyn Bridge<EventBus>>,
+    _wallet_updater: WalletUpdater,
+}
+
+pub enum Msg {
```

Comment:
TODO: For clarity we could rename this so that it's obvious that this message is _not_ sent to other components.